src/main/webapp/app/modules/canvas/visualization/visualization-properties/partials/visualization-data-properties.tsx

```diff
@@ -0,0 +1,239 @@
+                <ActionButton onPress={addFieldDimension} isQuiet={true}>
+                  <TableAndChart size="XXL" />
+                </ActionButton>
+                <Tooltip variant="info" showIcon>
+                  {'Add Dimension'}
+                </Tooltip>
+              </TooltipTrigger>
+            </Flex>
+            <Flex direction="column" gap="size-50" justifyContent="space-around">
+              <TooltipTrigger delay={0}>
+                <ActionButton onPress={addFieldMeasure} isQuiet={true}>
+                  <TableAndChart size="XXL" />
+                </ActionButton>
+                <Tooltip variant="info" showIcon>
+                  {'Add Measure'}
+                </Tooltip>
+              </TooltipTrigger>
+            </Flex>
+          </Flex>
+        </View>
+        <Form>         
+          <ButtonGroup orientation="vertical">
+            {props.visual.fields &&
+              props.visual.fields.length > 0 &&
+              props.visual.fields
+                .sort((a, b) => (a.fieldType.order > b.fieldType.order ? 1 : -1))
+                .map((field, i) => (
+                  <Button
+                    width={'100%'}
+                    marginTop={'size-25'}
+                    variant={selectedField?.fieldType?.id === field.fieldType.id ? 'cta' : 'secondary'}
+                    onPress={() => {
+                      selectedFieldChange(field);
+                    }}
+                    key={uuid()}
+                  >
+                    <div className={'field'}>
+                      <div className={'fieldKey'}>
+                        <div>
+                          <TableAndChart />
+                        </div>
+                        <div>
+                          <Text> {field.feature?.name || ''}</Text>
+                        </div>
+                      </div>
+                      <div>
+                        {field.fieldType.constraint === 'REQUIRED' && <LockClosed aria-label="Locked" />}
+                        {field.fieldType.constraint === 'OPTIONAL' && <Delete aria-label="Locked" />}
+                      </div>
+                    </div>
+                  </Button>
+                ))}
+          </ButtonGroup>
+
+          {showDimension && (
+            <Picker
+              onSelectionChange={selected => featureChange(selected.toString())}
+              label="Select dimension"
+              selectedKey={selectedField?.feature?.name || ''}
+            >
+              {dimensionList &&
+                dimensionList.length > 0 &&
+                dimensionList
+                  .sort((a, b) => (a.name > b.name ? 1 : -1))
+                  .map((feature, i) => <Item key={uuid() + '|' + feature.name}> {feature.name} </Item>)}
+            </Picker>
+          )}
+
+          {showMeasure && (
+            <Picker
+              onSelectionChange={selected => featureChange(selected.toString())}
+              label="Select measure"
```

Comment:
use translate function and move label into json file

Reply:
done